contractcourt/htlc_incoming_contest_resolver.go

```diff
@@ -201,23 +201,25 @@ func (h *htlcIncomingContestResolver) Resolve() (ContractResolver, error) {
 		HtlcID: h.htlc.HtlcIndex,
 	}
 
-	event, err := h.Registry.NotifyExitHopHtlc(
+	resolution, err := h.Registry.NotifyExitHopHtlc(
 		h.htlc.RHash, h.htlc.Amt, h.htlcExpiry, currentHeight,
 		circuitKey, hodlChan, payload,
 	)
-	switch err {
-	case channeldb.ErrInvoiceNotFound:
-	case nil:
-		defer h.Registry.HodlUnsubscribeAll(hodlChan)
-
-		// Resolve the htlc directly if possible.
-		if event != nil {
-			return processHtlcResolution(*event)
-		}
-	default:
+	if err != nil {
 		return nil, err
 	}
 
+	defer h.Registry.HodlUnsubscribeAll(hodlChan)
+
+	// If the event is non-nil, and the invoice is known to the registry
+	// (indicating that the htlc is not a forward), try to resolve the htlc
+	// directly.
+	if resolution != nil &&
```

Comment:
can both the resolution and err be nil?

Reply:
Yes, this comment should actually be re-phrased to say `if the resolution is non-nil`, will fix it now. 

We have a nil resolution and nil error when we are holding the htlc.